packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,562 @@
+  /// {@endtemplate}
+  double get itemExtent => _itemExtent;
+  double _itemExtent;
+  set itemExtent(double value) {
+    assert(value != null);
+    assert(value > 0.0);
+    if (value == _itemExtent)
+      return;
+    _itemExtent = value;
+    markNeedsLayout();
+  }
+
+  /// Whether to clip painted children to the inside of this viewport.
+  bool get clipToSize => _clipToSize;
+  bool _clipToSize;
+  set clipToSize(bool value) {
+    assert(value != null);
+    if (value == _clipToSize)
+      return;
+    _clipToSize = value;
+    _hasScrolled();
+  }
+
+  /// {@template flutter.rendering.wheelList.renderChildrenOutsideViewport}
+  /// Whether to paint children inside the viewport only.
+  ///
+  /// If false, every child will be painted. However the [Scrollable] is still
+  /// the size of the viewport and detects gestures inside only.
+  /// {@endtemplate}
+  bool get renderChildrenOutsideViewport => _renderChildrenOutsideViewport;
+  bool _renderChildrenOutsideViewport;
+  set renderChildrenOutsideViewport(bool value) {
+    assert(value != null);
+    if (value == _renderChildrenOutsideViewport)
+      return;
+    _renderChildrenOutsideViewport = value;
+    _hasScrolled();
+  }
+
+  void _hasScrolled() {
+    markNeedsPaint();
+    markNeedsSemanticsUpdate();
+  }
+
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! _ListWheelParentData)
+      child.parentData = new _ListWheelParentData();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _offset.addListener(_hasScrolled);
+  }
+
+  @override
+  void detach() {
+    _offset.removeListener(_hasScrolled);
+    super.detach();
+  }
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  /// Main axis length in the untransformed plane.
+  double get _viewportExtent {
+    assert(hasSize);
+    return size.height;
+  }
+
+  /// Main axis scroll extent in the **scrollable layout coordinates** that puts
+  /// the first item in the center.
+  double get _minScrollExtent {
+    assert(hasSize);
+    return 0.0;
+  }
+
+  /// Main axis scroll extent in the **scrollable layout coordinates** that puts
+  /// the last item in the center.
+  double get _maxScrollExtent {
+    assert(hasSize);
+    if (!(childCount > 0))
+      return 0.0;
+
+    return math.max(0.0, (childCount - 1) * _itemExtent);
+  }
+
+  /// Scroll extent distance in the untransformed plane between the center
+  /// position in the viewport and the top position in the viewport.
+  ///
+  /// It's also the distance in the untransformed plane that children's painting
+  /// is offset by with respect to those children's [BoxParentData.offset].
+  double get _topScrollMarginExtent {
+    assert(hasSize);
+    // Consider adding alignment options other than center.
+    return _minScrollExtent - size.height / 2.0 + _itemExtent / 2.0;
+  }
+
+  /// Transforms a **scrollable layout coordinates**' Y position to the
```

Comment:
prefer `y` to `Y` for consistency with other docs

Reply:
Done